ga4gh/datarepo.py

```diff
@@ -400,33 +412,50 @@ def _readSystemTable(self, cursor):
         self._creationTimeStamp = config["creationTimeStamp"]
 
     def _createOntologyTable(self, cursor):
+        # Right now we support a crude ontology term name-ID bidirectional
+        # map. However, in the future we will want to have better ontology
+        # support. Therefore we're not making the SQL schema too specific
+        # so that we can make this transition without needing backwards
+        # incompatible schema changes.
         sql = """
-            CREATE TABLE Ontology (
+            CREATE TABLE Ontology(
                 name TEXT NOT NULL PRIMARY KEY,
                 dataUrl TEXT NOT NULL
             );
         """
         cursor.execute(sql)
 
-    def insertOntology(self, ontology):
+    def insertOntologyTermMap(self, ontologyTermMap):
         """
-        Inserts the specified ontology into this repository.
+        Inserts the specified ontologyTermMap into this repository.
         """
         sql = """
-            INSERT INTO Ontology (name, dataUrl)
+            INSERT INTO Ontology(name, dataUrl)
             VALUES (?, ?);
         """
         cursor = self._dbConnection.cursor()
-        cursor.execute(sql, (
-            ontology.getLocalId(), ontology.getDataUrl()))
+        try:
+            cursor.execute(sql, (
+                ontologyTermMap.getLocalId(), ontologyTermMap.getDataUrl()))
+        except sqlite3.IntegrityError:
+            raise exceptions.DuplicateNameException(
+                ontologyTermMap.getLocalId())
 
-    def _readOntologyTable(self, cursor):
+    def _readOntologyTermMapTable(self, cursor):
         cursor.row_factory = sqlite3.Row
         cursor.execute("SELECT * FROM Ontology;")
         for row in cursor:
-            ontology = ontologies.Ontology(row[b'name'])
-            ontology.populateFromRow(row)
-            self.addOntology(ontology)
+            ontologyTermMap = ontologies.OntologyTermMap(row[b'name'])
+            ontologyTermMap.populateFromRow(row)
+            self.addOntologyTermMap(ontologyTermMap)
+
+    def removeOntologyTermMap(self, ontologyTermMap):
+        """
+        Removes the specified ontology term map from this repository.
+        """
+        sql = "DELETE FROM Ontology WHERE name=?"
+        cursor = self._dbConnection.cursor()
+        cursor.execute(sql, (ontologyTermMap.getLocalId(),))
```

Comment:
Yes, good point. We should at the very least offer the script that was used to generate it... @pcingola was the original contributor of this code, perhaps he has it sitting around somewhere.